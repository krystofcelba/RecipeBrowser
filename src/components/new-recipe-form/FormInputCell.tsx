import React from 'react';
import { View, StyleSheet } from 'react-native';
import { Cell } from 'react-native-tableview-simple';

import Input from './Input';

interface Props {
  placeholder: string;
  numberOfLines?: number;
  value: string;
  onChangeText: (text: string) => void;
}

const FormInputCell = (props: Props) => (
  <Cell
    cellContentView={
      <View style={styles.container}>
        <Input {...props} />
      </View>
    }
  />
);

const styles = StyleSheet.create({
  container: { flex: 1, paddingVertical: 10 },
});

export default FormInputCell;
