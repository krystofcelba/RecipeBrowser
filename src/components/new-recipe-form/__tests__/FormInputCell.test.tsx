import React from 'react';
import { shallow } from 'enzyme';

import FormInputCell from '../FormInputCell';

describe('FormInputCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<FormInputCell placeholder="placeholder" value="" onChangeText={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
