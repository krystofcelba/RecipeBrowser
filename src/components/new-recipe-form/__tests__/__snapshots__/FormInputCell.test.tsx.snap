// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputCell component renders as expected 1`] = `
<Cell
  accessory={false}
  accessoryColor="#007AFF"
  allowFontScaling={true}
  backgroundColor="#FFF"
  cellAccessoryView={null}
  cellContentView={
    <View
      style={
        Object {
          "flex": 1,
          "paddingVertical": 10,
        }
      }
    >
      <Input
        onChangeText={[Function]}
        placeholder="placeholder"
        value=""
      />
    </View>
  }
  cellImageView={null}
  cellStyle="Basic"
  contentContainerStyle={Object {}}
  detail=""
  detailTextStyle={Object {}}
  disableImageResize={false}
  highlightActiveOpacity={0.8}
  highlightUnderlayColor="black"
  image={null}
  isDisabled={false}
  leftDetailColor="#007AFF"
  onHighlightRow={null}
  onPress={false}
  onUnHighlightRow={null}
  rightDetailColor="#8E8E93"
  subtitleColor="#000"
  subtitleTextStyle={Object {}}
  title=""
  titleTextColor="#000"
  titleTextStyle={Object {}}
  titleTextStyleDisabled={Object {}}
/>
`;
